hspec-core/src/Test/Hspec/Core/Runner.hs

```diff
@@ -122,13 +123,17 @@ hspecWithResult conf spec = do
         seed = (fromJust . configQuickCheckSeed) c
         qcArgs = configQuickCheckArgs c
 
+    jobsSem <- case configMaxParallelJobs c of
+      Nothing -> return Nothing
```

Comment:
In case `-j` is omitted can we use `getNumCapabilities` as a default? That would mean that we *always* use a semaphore, sometimes with a capacity of 1. This should hopefully simplify the code.